boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,346 @@
+    efuse_context->channel0_port   = channel0_port;
+    efuse_context->channel0_pin    = channel0_pin;
+    efuse_context->channel1_port   = channel1_port;
+    efuse_context->channel1_pin    = channel1_pin;
+    efuse_context->wdin_bit_to_set = true;
+
+    return efuse_context;
+}
+
+void Io_Efuse_Destroy(struct Efuse_Context *efuse)
+{
+    free(efuse);
+}
+
+void Io_Efuse_EnableChannel0(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel0(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(
+        efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_RESET);
+}
+
+void Io_Efuse_EnableChannel1(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel1(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(
+        efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     monitoring_function,
+    struct Efuse_Context *const efuse)
+{
+    uint16_t register_value;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_ReadRegister(SI_GCR_ADDR, &register_value, efuse));
+
+    CLEAR_BIT(register_value, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(
+        register_value,
+        (monitoring_function & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, register_value, efuse));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_ExitFailSafeMode(struct Efuse_Context *const efuse)
+{
+    // Set the WDIN bit (15th bit) to exit out of fail-safe mode, the parity bit
+    // (14th bit) will automatically be set by the Io_Efuse_WriteRegister()
+    // function 1_1_00000_00000_0000
+    efuse->wdin_bit_to_set = true;
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_STATR_0_ADDR, 0x0000, efuse));
+
+    // Disable the watchdog timer
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, GCR_CONFIG, efuse));
+
+    // Check if the the efuse is still in fail-safe mode
+    if (Io_Efuse_IsEfuseInFailSafeMode(efuse))
+    {
+        return EXIT_CODE_UNIMPLEMENTED;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+bool Io_Efuse_IsEfuseInFaultMode(const struct Efuse_Context *const efuse)
+{
+    return HAL_GPIO_ReadPin(efuse->fsb_port, efuse->fsb_pin);
+}
+
+bool Io_Efuse_IsEfuseInFailSafeMode(const struct Efuse_Context *const efuse)
+{
+    return HAL_GPIO_ReadPin(efuse->fsob_port, efuse->fsob_pin);
+}
+
+void Io_Efuse_DelatchFaults(const struct Efuse_Context *const efuse)
+{
+    // Delatch the latchable faults by alternating channel 0's and channel 1's
+    // input pins high->low->high.
+    HAL_GPIO_WritePin(efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_SET);
```

Comment:
Does this need to be completed in the particular sequence? Or can you do something like this
```
//Set pins high
HAL_WritePin(Channel0);
HAL_WritePin(Channel1);

for(i = 0; i < 2; i++)
{
   HAL_TogglePin(Channel0);
   HAL_TogglePin(Channel1);
}

```

Reply:
Yes, it need to be done in the sequence as it is currently done. Something as simple as setting/resetting a GPIO pin should be kept simple. No need for a for loop, this also add overhead. I don't see what you gain from this?